library/firewall_lib.py

```diff
@@ -178,7 +178,39 @@
 except ImportError:
     HAS_FIREWALLD = False
 
-
+def handle_interface(zone,item,fw_zone,fw_settings,fw,fw_offline):
+    if fw_offline:
+        iface_zone_objs = []
+        for zone in fw.config.get_zones():
+            old_zone_obj = fw.config.get_zone(zone)
+            if item in old_zone_obj.interfaces:
+                iface_zone_objs.append(old_zone_ob)
+        if len(iface_zone_objs) > 1:
+            module.fail_json(
+                msg="ERROR: interface {0} is in {1} zone XML file, can only be in one".format(
+                    interface, len(iface_zone_objs)
+                )
+            )
```

Comment:
This should never happen. If we want to enforce the "an interface is bound to one and only one zone" rule in this role then why can't we just remove the interface from all zones other than the new zone? No need to throw an error and give up. This will also simplify the surrounding code.